### ISSUE TITLE
Support all tar options in archive state

### ISSUE THREAD

**Issue by rchrd** [saltstack/salt#10293]
The archive states extracted method has the attribute 'tar_options' but this inserts the option in '-xv{0}f'. Therefor it is not possible to use an option like --strip-components=1. This options is very useful to skip the rootlevel of an archive so it is extracted in a known location.

For example the tarfile of a JBoss server creates a directoryname with the version number in it. This makes it difficult to reference (and control) files within that structure.

So changing the line:
```
results = __salt__['cmd.run_all']('tar -xv{0}f {1}'.format(tar_options, filename), cwd=name)
```
to
```
results = __salt__['cmd.run_all']('tar -xvf {1} {0}'.format(tar_options, filename), cwd=name)
```
would help a lot.

**Comment by frumpel** [comment]
Don't have the time to test rigorously right now but I had a working recipe in Hydrogen that did this

```
- tar_options: "-xz --strip-components=1"
```

In Helium (2014.7) this appears to have regressed with undesirable results. It appears this is now converted to:

```
[INFO    ] Executing command ['tar', 'x-xz --strip-components=1', '-f', "'/var/cache/salt/minion/files/base/foo.tar'"] in directory 'bar'
```

**Comment by frumpel** [comment]
The truly annoying bit is that the tar options are being quoted! So this will fail too, now because tar misinterprets things:

```
- tar_options: "-xz --strip-components=1"
```

**Comment by frumpel** [comment]
I would suggest that tar_options should accept an ARRAY as input. That way you can pass an arbitrary set of parameters. Ok, it's not generic but it's better than the alternative:

* download tar file into explicit temp directory
* use cmd.run to expand with a "require" dependency on the file
* do this every time or incur a failure because if you put the same "unless extracted bits already exist" on both the file download and the cmd.run then this will lead to a "failure" of the tar command once the file has been extracted (another bug about ordering of require and unless, forgot the number)

**Comment by bbinet** [comment]
I believe this is actually fixed since 2015.5: I'm now able to use successfully `- tar_options: "--strip-components=1"`
See also #19799.

**Comment by basepi** [comment]
Awesome! In that case, I'm going to close unless someone thinks this is not fixed.
